server/src/test/java/org/elasticsearch/common/xcontent/BaseXContentTestCase.java

```diff
@@ -1007,22 +1008,20 @@ public void testNamedObject() throws IOException {
         {
             p.nextToken();
             assertEquals("test", p.namedObject(Object.class, "str", null));
-            UnknownNamedObjectException e = expectThrows(UnknownNamedObjectException.class,
+            XContentParseException e = expectThrows(XContentParseException.class,
```

Comment:
I think this should be `NamedObejectNotFoundException` instead, right? It is more specific.